src/binaryen-c.h

```diff
@@ -64,13 +64,14 @@ typedef uint32_t BinaryenIndex;
 // Core types (call to get the value of each; you can cache them, they
 // never change)
 
-typedef uint32_t BinaryenType;
+typedef int32_t BinaryenType;
```

Comment:
this should still be unsigned. might require writing `uint32_t(-1)` if the compiler complains.